LDMP/api.py

```diff
@@ -126,324 +144,339 @@ def finished(self, result):
+            # Strip password out of payload for printing to QGIS logs
 
-def call_api(endpoint, method="get", payload=None, use_token=False):
-    if use_token:
-        token = login()
+            if payload:
+                clean_payload = self._clean_payload(payload)
+            else:
+                clean_payload = payload
+            log('API calling {} with method "{}"'.format(endpoint, method))
 
-        if token:
             if conf.settings_manager.get_value(conf.Setting.DEBUG):
-                log("API loaded token.")
-            headers = {"Authorization": f"Bearer {token}"}
-        else:
-            return
-    else:
-        if conf.settings_manager.get_value(conf.Setting.DEBUG):
-            log("API no token required.")
-        headers = {}
-
-    # Only continue if don't need token or if token load was successful
-
-    if (not use_token) or token:
-        # Strip password out of payload for printing to QGIS logs
+                log("API call payload: {}".format(clean_payload))
+            resp = self._make_request(
+                "Trends.Earth API call",
+                url=self.url + endpoint,
+                method=method,
+                payload=payload,
+                headers=headers,
+                timeout=self.timeout,
+            )
 
-        if payload:
-            clean_payload = _clean_payload(payload)
         else:
-            clean_payload = payload
-        log('API calling {} with method "{}"'.format(endpoint, method))
-
-        if conf.settings_manager.get_value(conf.Setting.DEBUG):
-            log("API call payload: {}".format(clean_payload))
-        resp = _make_request(
-            "Trends.Earth API call",
-            url=API_URL + endpoint,
-            method=method,
-            payload=payload,
-            headers=headers,
-        )
-
-    else:
-        resp = None
+            resp = None
 
-    if resp != None:
-        if resp.status_code == 200:
-            ret = resp.json()
+        if resp != None:
+            if resp.status_code == 200:
+                ret = resp.json()
+            else:
+                desc, status = self.get_error_status(resp)
+                err_msg = "Error: {} (status {}).".format(desc, status)
+                log(err_msg)
+                """
+                iface.messageBar().pushCritical(
+                    "Trends.Earth", "Error: {} (status {}).".format(desc, status)
+                )
+                """
+                ret = None
         else:
-            desc, status = get_error_status(resp)
-            err_msg = "Error: {} (status {}).".format(desc, status)
-            log(err_msg)
-            """
-            iface.messageBar().pushCritical(
-                "Trends.Earth", "Error: {} (status {}).".format(desc, status)
-            )
-            """
             ret = None
-    else:
-        ret = None
 
-    return ret
+        return ret
 
+    def get_header(self, url):
```

Comment:
This function doesn't actually use the API, so would make more sense to handle this in a different class (it is for getting the header of any generic URL, needed for checking file size, etc.)